### PR TITLE
Feature/school-292/invite-parent

### PR DESCRIPTION
**Pull request by Ivan228777**
Added invite-parent and invite-parent-invitations-list API handlers, covered them with tests.

### CODE REVIEW

File: open_schools_platform/common/services.py [hunk 1]
```diff
@@ -60,12 +60,15 @@ def query_handler(query: Query, new_status: str, user: User):
...
-    def query_handler_checks(query_handler_class, query: Query, new_status: str, user: User):
+    def query_handler_checks(query_handler_class, query: Query, new_status: str, user: User, without_body: bool = None):
         if new_status not in query_handler_class.allowed_statuses:
             raise NotAcceptable("Not allowed status")
         if query.status == new_status:
             raise ValidationError(detail="Identical statuses")
-        if query.recipient is None or query.sender is None or query.body is None:
+        if query.recipient is None or query.sender is None:
+            if not without_body:
+                if query.body is None:
```
Comment (zavarin-michael) [inline open_schools_platform/common/services.py:70]: you can write like this:
not without_body and query.body is None

**Review by zavarin-michael — CHANGES_REQUESTED** [review]

File: open_schools_platform/query_management/queries/models.py [hunk 2]
```diff
@@ -27,13 +27,13 @@ class Status(models.TextChoices):
 
     sender = GenericForeignKey("sender_ct", "sender_id")
 
-    body_ct = models.ForeignKey(ContentType, related_name="body_ct", null=True, on_delete=models.CASCADE)
+    body_ct = models.ForeignKey(ContentType, related_name="body_ct", null=True, blank=True, on_delete=models.CASCADE)
```
[the remark this answers is not in the document]
Comment (Ivan228777) [reply open_schools_platform/query_management/queries/models.py:30]: We need it for making empty body in Query

File: open_schools_platform/query_management/queries/tests/utils.py [hunk 3]
```diff
@@ -47,6 +49,16 @@ def create_test_employee_invite_organization_query(user: User = None,
...
 
 
+def create_test_family_invite_parent_query(i: int = 0, family: Family = None, user: User = None) -> Query:
+    if user is None:
+        user = create_test_user(phone="+79021111111")
+    if family is None:
+        family = create_test_family(i=1, parent=create_test_user(phone="+79022222222").parent_profile)
+    query = create_query(sender_model_name="family", sender_id=family.id,
+                         recipient_model_name="parentprofile", recipient_id=user.parent_profile.id)
+    return query
```
[the remark this answers is not in the document]
Comment (Ivan228777) [reply open_schools_platform/query_management/queries/tests/utils.py:59]: Also we need this variable for proper work of create_testdata_in_db function.

File: open_schools_platform/query_management/queries/models.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (zavarin-michael) [reply open_schools_platform/query_management/queries/models.py:30]: it is strange, this comment was deleted by me
okay, understandable

File: open_schools_platform/query_management/queries/tests/utils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (zavarin-michael) [reply open_schools_platform/query_management/queries/tests/utils.py:59]: okay but rename it
object_index_get or or something like that

File: open_schools_platform/common/services.py [hunk 4]
```diff
@@ -65,10 +65,7 @@ def query_handler_checks(query_handler_class, query: Query, new_status: str, use
             raise NotAcceptable("Not allowed status")
         if query.status == new_status:
             raise ValidationError(detail="Identical statuses")
-        if query.recipient is None or query.sender is None:
-            if not without_body:
-                if query.body is None:
-                    raise MethodNotAllowed("put", detail="Query is corrupted")
+        if query.recipient is None or query.sender is None or without_body is None and query.body is None:
```
Comment (zavarin-michael) [inline open_schools_platform/common/services.py:68]: why without_body is None?)
bool should be false or true

**Review by zavarin-michael — CHANGES_REQUESTED** [review]

**Review by zavarin-michael — APPROVED** [review]

**Pull request merged**